### PR TITLE
Define strong and weak ETags separately

### PR DESCRIPTION
**Pull request by josh-newman**
@saeta

### CODE REVIEW

File: naptime/src/main/scala/org/coursera/naptime/ETag.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+object ETag {
+
+  implicit val stringKeyFormat: StringKeyFormat[ETag] = {
+    StringKeyFormat.unimplementedFormat[ETag]
+      .orFormat[Strong]
+      .orFormat[Weak]
+  }
+
+  /**
+   * TODO(josh): Should Naptime support this at all? Play JSON serialization doesn't allow
```
Comment (saeta) [inline naptime/src/main/scala/org/coursera/naptime/ETag.scala]: Could you actually just switch everything to use the weak etags? I believe that is the most semantically correct thing to do.

File: naptime/src/main/scala/org/coursera/naptime/ETag.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.coursera.common.stringkey.StringKeyFormat
+import org.coursera.common.stringkey.StringKeyFormat.Implicits.OrFormat
+
+/**
+ * Note: Naptime always uses weak validators because the framework does not guarantee byte-equality
+ * between responses.
+ *
+ * @param weakValidator tag without surrounding `"`s; these are added by Naptime
+ */
+case class ETag(weakValidator: String)
```
Comment (saeta) [inline naptime/src/main/scala/org/coursera/naptime/ETag.scala]: Should we require that `weakValidator` does not contain a `"`?

**Comment by saeta** [comment]
@josh-newman looks like there are some important test failures. Could you update them? After that, then LGTM from me.

**Comment by josh-newman** [comment]
@saeta, PTAL.

**Comment by saeta** [comment]
@josh-newman LGTM.

**Pull request merged**